internal/services/compute/virtual_machine_scale_set_data_source.go

```diff
@@ -164,6 +164,9 @@ func dataSourceVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, meta interf
 		if instance.InstanceID != nil {
 			nics, err := networkInterfacesClient.ListVirtualMachineScaleSetVMNetworkInterfacesComplete(ctx, id.ResourceGroup, id.Name, *instance.InstanceID)
 			if err != nil {
+				if utils.ResponseWasNotFound(nics.Response().Response) {
+					break
+				}
```

Comment:
since a VM has to have a NIC attached, what's the scenario that causes this? Feels like something we should add a test for?

Reply:
This has to do with using this data source to import an orchestrated vmss

```
=== CONT  TestAccDataSourceVirtualMachineScaleSet_orchestrated
testcase.go:110: Step 1/1 error: Error running post-apply plan: exit status 1
Error: listing Network Interfaces for VM Instance "acctestvmonxpc" for Virtual Machine Scale Set "acctestRG-230315042426586344" (Resource Group "acctestVMO-230315042426586344"): network.InterfacesClient#ListVirtualMachineScaleSetVMNetworkInterfaces: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="NotFound" Message="Resource /subscriptions/*******/resourceGroups/acctestRG-230315042426586344/providers/Microsoft.Compute/virtualMachineScaleSets/acctestVMO-230315042426586344 not found." Details=[]
with data.azurerm_virtual_machine_scale_set.test,
on terraform_plugin_test.tf line 94, in data "azurerm_virtual_machine_scale_set" "test":
94: data "azurerm_virtual_machine_scale_set" "test" {
--- FAIL: TestAccDataSourceVirtualMachineScaleSet_orchestrated (385.82s)
FAIL
```

Reply:
that'd be because this is the wrong API to call for an Orchestrated VMSS - since they're so different we should likely introduce a data source for the Orchestrated VMSS tbh?

Reply:
That's fair! I'll look into it